dipy/direction/tests/test_peaks.py

```diff
@@ -690,3 +688,49 @@ def test_reshape_peaks_for_visualization(rng):
     assert_array_equal(data1_reshape.reshape(10, 5, 3), data1)
     assert_array_equal(data2_reshape.reshape(10, 2, 5, 3), data2)
     assert_array_equal(data3_reshape.reshape(10, 2, 12, 5, 3), data3)
+
+
+def test_peaks_from_positions():
+
+    thresh = .5
+    min_angle = 25
+    npeaks = 5
+
+    _, fbvals, fbvecs = get_fnames('small_64D')
+    bvals, bvecs = read_bvals_bvecs(fbvals, fbvecs)
+    gtab = gradient_table(bvals, bvecs)
+    model = CsaOdfModel(gtab, 4)
+    mevals = np.array(([0.0015, 0.0003, 0.0003],
+                       [0.0015, 0.0003, 0.0003]))
+    voxels = []
+    for i in range(27):
+        v, _ = multi_tensor(gtab, mevals, S0=100,
+                            angles=[(0, 0), (randint(0, 90), randint(0, 90))],
+                            fractions=[50, 50], snr=10)
+        voxels.append(v)
+    data = np.array(voxels).reshape((3, 3, 3, -1))
+
+    pam = peaks_from_model(model, data, default_sphere, return_odf=True,
```

Comment:
When we run the tests, a warning appear. You need to filter this warning with
```
with warnings.catch_warnings():
            warnings.filterwarnings(
                "ignore", message=descoteaux07_legacy_msg,
                category=PendingDeprecationWarning)
```

See line 466 above, on the same file

Reply:
There is an issue here. 

`Legacy=True` in my call to `peaks_from_model(.)`. However, the is no way to set the `Legacy=True` flag when creating a Dipy model (e.g. `model = CsaOdfModel(gtab, 4)`). Thus, when setting `Legacy=True` in `peaks_from_model`,  ODFs will always be wrongly scaled. I.e. the legacy option is set both in the model object and in `peaks_from_model`, if they are not equal, the results will be wrong.

TLDR: `peaks_from_model(Legacy=True)` will always output wrongly scaled ODFs.

Am I getting something wrong? @skoudoro @Garyfallidis

Reply:
To solve this, we could remove the `Legacy` parameter from `peaks_from_model`, and add it to all SH models. peaks_from_model should get the legacy option from the model parameter. SH models should set it correctly on object creation.

Reply:
fixed in this PR by setting `Legacy=True`

Reply:
My bad, the code is fine. The code go back to the ODF then to the selected SH basis.

Reply:
I created issue #3187 to add support for `Legacy=False` for SH basis model.